Makefile

```diff
@@ -149,10 +155,11 @@ build-node-image: buildx-setup docker-pull-prerequisites ## Build node-manager i
 		--tag $(NODE_MANAGER_LINUX_FULL_IMAGE):$(IMAGE_TAG)-$(ARCH) .
 
 .PHONY: build-and-push-node-image-windows
-build-and-push-node-image-windows: buildx-setup ## Build node-manager image for Windows and push it to registry.
-	go build -a -o $(BIN_DIR)/azure-cloud-node-manager.exe ./cmd/cloud-node-manager
-	docker buildx build --pull --push --output=type=registry --platform windows/amd64 \
-		-t $(NODE_MANAGER_WINDOWS_IMAGE)-$(WINDOWS_OSVERSION) --build-arg OSVERSION=$(WINDOWS_OSVERSION) \
+build-and-push-node-image-windows: buildx-setup $(BIN_DIR)/azure-cloud-node-manager.exe ## Build node-manager image for Windows and push it to registry.
```

Comment:
could we split `build-and-push` into `build` and `push` for windows image?

Reply:
I have refactored azure file Makefile multiple times to make it clean, feel free to copy the logic

Reply:
It's possible to split the build-and-push for windows image, but in some sense, it makes no sense, we cannot always build windows image on a Linux unless direct the output to local file. will do it anyway @luncj 

I can take a reference from azure file, thanks for your work @andyzhangx

Reply:
Borrowed some lines of code from azurefile, please take another look.